Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeAMSIConfigurationState.ps1

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+. $PSScriptRoot\..\..\Helpers\Invoke-CatchActions.ps1
+Function Get-ExchangeAMSIConfigurationState {
+
+    begin {
+        Write-Verbose "Calling: $($MyInvocation.MyCommand)"
+    } process {
+        try {
+            Write-Verbose "Trying to query AMSI configuration state"
+            $amsiConfiguration = Get-SettingOverride | Where-Object { ($_.ComponentName -eq "Cafe") -and ($_.SectionName -eq "HttpRequestFiltering") } -ErrorAction Stop
+
+            if (($null -ne $amsiConfiguration) -and
+                ($amsiConfiguration.Count -eq 1)) {
+                Write-Verbose "Setting override detected for AMSI configuration"
+                $amsiConfigurationQuerySuccessful = $true
+                Switch ($amsiConfiguration.Parameters.Split("=")[1]) {
+                    ("False") { $amsiState = $false }
+                    ("True") { $amsiState = $true }
+                    Default { $amsiState = "Unknown" }
+                }
+
+                if ($null -eq $amsiConfiguration.Server) {
+                    $amsiOrgWideSetting = $true
+                } else {
+                    $amsiOrgWideSetting = $false
+                }
+            } elseif (($null -ne $amsiConfiguration) -and
+                ($amsiConfiguration.Count -gt 1)) {
```

Comment:
How do we get multiple overrides? How do we determine which one is active in Exchange, might need to change this logic

Reply:
It's possible to configure multiple overrides by using different IDs but the same component / section combination. Not sure how Exchange handles this case but it's possible to create these conflicting configurations (validated in lab). I have no idea yet how to handle these cases at best so the current idea is to write a warning that we cannot determine if the configuration is accurate or not. But agree, we should re-think this part and find a better solution in the near future.

Reply:
I have no good idea how to better handle this case. It's possible to have multiple and conflicting settings:
![image](https://user-images.githubusercontent.com/40993616/142454990-bf9e8e74-83c7-49c4-9d70-62b8c61908e1.png)
![image](https://user-images.githubusercontent.com/40993616/142455126-3bfd671f-ed13-4f2e-a26b-673c41369d62.png)

I don't think that we see this scenario that much and so would like to start with this simple logic to call out that we've detected multiple settings and so cannot clearly determine the state of the AMSI Exchange integration:
![image](https://user-images.githubusercontent.com/40993616/142455413-ec82c989-ae69-4b9f-a7cc-b10e61742e1d.png)

What do you think?

Reply:
Should add this to the AMSI check description with a few sentence about how to check and delete conflicting settings.

Reply:
I think for this, we need to just return a list of all the setting overrides and loop through them and display them all with the Health Checker. Still call out the Multiple overrides detected, but just display it the critical information that we want for each setting.

Reply:
Yeah, might confuse the user but should be no problem to add the information.

Reply:
Re-worked the multiple override detection / output. It now looks like this:
<img width="488" alt="image" src="https://user-images.githubusercontent.com/40993616/142655709-90cd677c-efe0-465f-9811-60038a9ddb9c.png">

Reply:
Added with latest commit. Resolve this conversation now. Please re-open if something else should be added.